### ISSUE TITLE
Tensor size error in MedNextv1 trainer

### ISSUE THREAD

**Issue by mariaperezf** [MIC-DKFZ/MedNeXt#34]
Hello, I am having trouble with the following Runtime Error:

`RuntimeError: The size of tensor a (3) must match the size of tensor b (4) at non-singleton dimension 2`

When trying to train a model using MedNext with the command:

`mednextv1_train 3d_fullres nnUNetTrainerV2_MedNeXt_S_kernel3 999 0`

**Comment by Helen52520** [comment]
Got the same error. Did you fix the problem? Thanks!

**Comment by mariaperezf** [comment]
> Got the same error. Did you fix the problem? Thanks!

No, not really. Although it does work if you use the preprocessing from the MICCAI 2023 that is shown in the READEME, so I guess it has something to do with the spacing:

mednextv1_plan_and_preprocess -t YOUR_TASK -pl3d ExperimentPlanner3D_v21_customTargetSpacing_1x1x1 -pl2d ExperimentPlanner2D_v21_customTargetSpacing_1x1x1

mednextv1_train 3d_fullres TRAINER TASK_NUMBER FOLD -p nnUNetPlansv2.1_trgSp_1x1x1

**Comment by Helen52520** [comment]
> > Got the same error. Did you fix the problem? Thanks!
> 
> No, not really. Although it does work if you use the preprocessing from the MICCAI 2023 that is shown in the READEME, so I guess it has something to do with the spacing:
> 
> mednextv1_plan_and_preprocess -t YOUR_TASK -pl3d ExperimentPlanner3D_v21_customTargetSpacing_1x1x1 -pl2d ExperimentPlanner2D_v21_customTargetSpacing_1x1x1
> 
> mednextv1_train 3d_fullres TRAINER TASK_NUMBER FOLD -p nnUNetPlansv2.1_trgSp_1x1x1

I successfully got it running with your command, thank you! By the way, may I ask if you're also working with multi-modality? The original command (-p nnUNetPlansv2.1) works for single-channel, but it fails for multi-modality.

**Comment by mariaperezf** [comment]
> > > Got the same error. Did you fix the problem? Thanks!
> > 
> > 
> > No, not really. Although it does work if you use the preprocessing from the MICCAI 2023 that is shown in the READEME, so I guess it has something to do with the spacing:
> > mednextv1_plan_and_preprocess -t YOUR_TASK -pl3d ExperimentPlanner3D_v21_customTargetSpacing_1x1x1 -pl2d ExperimentPlanner2D_v21_customTargetSpacing_1x1x1
> > mednextv1_train 3d_fullres TRAINER TASK_NUMBER FOLD -p nnUNetPlansv2.1_trgSp_1x1x1
> 
> I successfully got it running with your command, thank you! By the way, may I ask if you're also working with multi-modality? The original command (-p nnUNetPlansv2.1) works for single-channel, but it fails for multi-modality.

No I am not working multi-modality, sorry! I have been working with CT only.  However, even though the training was suscessful I have not been able to do the inference with the test model with the trained MedNeXt. When tryining to use the nnUNet_predict command I get the error:

`RuntimeError: Could not find the model trainer specified in checkpoint in nnunet.trainig.network_training. If it is not located there, please move it or change the code of restore_model. Your model trainer can be located in any directory within nnunet.trainig.network_training (search is recursive).`
